docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -20,401 +51,355 @@ The scenarios of defining placement rules in SQL include:
+For example, `FOLLOWER_CONSTRAINTS="[+region=us-east-1,+region=us-east-2]" FOLLOWERS=3` indicates to place 3 followers on either `us-east-1` or `us-east-1`. There may be 2 replicas on `us-east-1` and 1 in `us-east-2`, or 2 in `us-east-2` and 1 in `us-east-1`. It's up to PD, but PD will attempt to balance such that 3 do not end up in one region (TODO: this conflicts with "Ambiguous and edge cases" below, need to confirm expected behavior).
 
 Label constraints can be implemented by defining `label_constraints` field in PD placement rule configuration. `+` and `-` correspond to property `op`. Specifically, `+` is equivalent to `in` and `-` is equivalent to `notIn`.
 
-For example, `+zone=sh,+zone=bj,-disk=hdd` is equivalent to:
+For example, `+region=us-east-1,+region=us-east-2,-disk=hdd` is equivalent to:
 
 ```
 "label_constraints": [
-	{"key": "zone", "op": "in", "values": ["sh", "bj"]},
+	{"key": "region", "op": "in", "values": ["us-east-1", "us-east-2"]},
 	{"key": "disk", "op": "notIn", "values": ["hdd"]}
 ]
 ```
 
 Field `location_labels` in PD placement rule configuration is used to isolate replicas to different zones to improve availability. For now, the global configuration can be used as the default `location_labels` for all placement rules defined in SQL, so it's unnecessary to specify it.
 
-### Role configuration
-
-`ROLE` in the statement defines the Raft role of the replicas. It must be specified in the statement. There are 4 predefined roles:
-
-- `leader`. Exactly one `leader` is allowed.
-- `follower`.
-- `voter`. It includes `leader` and `follower`.
-- `learner`. It can be either TiFlash or TiKV.
-
-If both `voter` and `follower` are defined in the rules, the replicas of `follower` are not included in the replicas of `voter`. For example:
+`PLACEMENT` also supports adding TiFlash replicas for a table, as statement `ALTER TABLE table_name SET TIFLASH REPLICA count` does. For example:
 
 ```sql
-ALTER TABLE test
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=bj]" ROLE=follower REPLICAS=2,
-	ALTER PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=voter REPLICAS=2;
+ALTER TABLE t1
+	LEARNER_CONSTRAINTS="[+engine=tiflash]" LEARNERS=1;
 ```
 
-There are 4 replicas for table `test`, 2 of which are in `sh` and 2 are in `bj`.  Leader can only be placed on `sh`.
+The only way to judge whether it’s adding a TiFlash replica is to check the label. If it contains `engine=tiflash`, then it’s adding or removing a TiFlash replica. This logic is conventional in PD for now.
 
-`ROLE` in the statement is equivalent to field `role` in PD placement rule configuration.
+#### Specifying role count
 
-### Replicas configuration
+The roles `FOLLOWERS`, `LEARNERS` and `VOTERS` also support an optional count in *list* format. For example:
 
-`REPLICAS` in the statement indicates the replica count of the specified role.
+```sql
+CREATE PLACEMENT POLICY `standardplacement1` PRIMARY REGION="us-east-1" REGIONS="us-east-1,us-east-2" FOLLOWERS=4;
```

Comment:
I think we need to clarify the meaning of PRIMARY, which one？
-  the leader is placed in the Region (improve read)
-  the leader is placed in the Region, and the majority of replicas are placed in the primary region (improve read and write)

Reply:
In addition, maybe we need to comment that if you do not specify replicas number, then the number of the replicas 
 inherited from up level.

Reply:
@nolouch I think that it means the leader is placed in that region. Whether the majority of replicas are placed there as well depends on the `SCHEDULE` (i.e. what the balancing goal is).

So one `SCHEDULE` should imply most are local for fast quorum, and another might aim for balanced across as many regions as possible.

Reply:
This has been addressed in https://github.com/pingcap/tidb/pull/26221/commits/af6c0ce8fa2a9f52aba4282c9f74d6719ba37fa9